lib/api/BackbeatAPI.js

```diff
@@ -615,6 +622,156 @@ class BackbeatAPI {
+     * @param {object} details - route details from lib/backbeat/routes.js
+     * @param {function} cb - callback(error, data)
+     * @param {array} data - optional field providing already fetched data in
+     *   order of dataPoints mentioned for each route in lib/backbeat/routes.js
+     * @return {undefined}
+     */
+    getFailedMetrics(details, cb, data) {
+        const { failedCRR } = redisKeys;
+        const site = details.site === 'all' ? '*' : details.site;
+        const pattern = `${failedCRR}:*:${site}:*`;
+
+        this._redisClient.scan(pattern, undefined, (err, keys) => {
+            if (err) {
+                this._logger.error('error getting metric: failures', {
+                    error: err,
+                    method: 'BackbeatAPI.getFailMetrics',
+                });
+                return cb(errors.InternalError);
+            }
+            const filteredKeys = this._filterFailObjectKeys(keys);
+            const opsFail = filteredKeys.length;
+            // TODO: for now, not returning failures bytes. Instead, rely on
+            // retry to return obj sizes
+            const bytesFail = 0;
+
+            const uptime = this._getMaxUptime(EXPIRY);
+            const response = {
+                failures: {
+                    description: 'Number of failed replication operations ' +
+                        '(count) and bytes (size) in the last ' +
+                        `${Math.floor(uptime)} seconds`,
+                    results: {
+                        count: opsFail,
+                        size: bytesFail,
+                    },
+                },
+            };
+            return cb(null, response);
+        });
+    }
+    /* eslint-enable */
+
+    _filterFailObjectKeys(keys) {
+        const filteredKeys = [];
+        const uniqVersionIDs = {};
+        const removeKeys = [];
+        keys.forEach(key => {
+            // 8 parts: [bb, crr, failed, <bucket>, <key>,
+            //           <versionID>, <location>, <timestamp>]
+            const parts = key.split(':');
+            const versionID = process.env.CI === 'true' ? parts[6] : parts[5];
+            if (versionID === '') {
+                filteredKeys.push(key);
+                return;
+            }
+            // if the versionID already exists, debug log because an error
+            // occurred somewhere
+            if (uniqVersionIDs[versionID]) {
+                this._logger.debug('duplicate version id was found in failed ' +
+                'object redis keys', {
+                    method: 'BackbeatAPI._filterFailObjectKeys',
+                    message: 'Known bug with unknown origin. Debug and ' +
+                        'report this behavior please',
+                });
+                removeKeys.push(key);
+            } else {
+                uniqVersionIDs[versionID] = key;
+            }
+        });
+        filteredKeys.push(...Object.keys(uniqVersionIDs)
```

Comment:
I think we can avoid this loop by just pushing the key to `filteredKeys` in the `else` block above.